File: CSharpCodeBase/aiactions/ai_timeout.lua
local Class = require 'utils.hump.class'
local AINode = require 'utils.ailib.ainode'

-- Select enemy using ai vision
local AI_Timeout = Class({__includes=AINode, init=function(self, entity, timeout)
  AINode.init(self, entity)
  self.timeout = timeout
  self.currentTimeout = timeout
end})

function AI_Timeout:Visit()
  if self.status == NODE_READY then
    self.status = NODE_RUNNING
    self.currentTimeout = self.timeout
  end
  if self.status == NODE_RUNNING  then
    --[[
    if self.parent.selected_target ~= nil then
      self.entity.mover:LookAt(self.parent.selected_target:GetPosition())
    end
    ]]
    self.currentTimeout = self.currentTimeout - GameController.deltaTime
    if self.currentTimeout <= 0 then
      self.status = NODE_SUCCESS
    end
  end
  return self.status
end

return AI_Timeout

File: CSharpCodeBase/aiactions/ai_timeout.cs
using System;
 
 using System.Collections.Generic;
 
 using System.Linq;
 
 using System.Text;
 
 using UnityEngine;
 
 using System.Collections;
 
 using MainGame.core;
 
 using UnityEngine.EventSystems;
 
 namespace MainGame{
 public class ai_timeout :ainode{
 // Select enemy using ai vision
 public void init(self, entity, timeout);
   AINode.init(self, entity);
   this.timeout = timeout;
   this.currentTimeout = timeout;
 }})
 public void Visit(){
   if(this.status == NODE_READY  ){ 
     this.status = NODE_RUNNING;
     this.currentTimeout = this.timeout;
   }
   if(this.status == NODE_RUNNING   ){ 
     ////
     if(this.parent.selected_target != null  ){ 
       this.entity.mover:LookAt(this.parent.selected_target:GetPosition());
     }
     //
     this.currentTimeout = this.currentTimeout - GameController.deltaTime;
     if(this.currentTimeout <= 0  ){ 
       this.status = NODE_SUCCESS;
     }
   }
   return this.status;
 }
 }}